src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -883,16 +883,14 @@ sequence _f _p = undefined
 -------------------------------------------------------------------------------
 -- Alternative Collection
 -------------------------------------------------------------------------------
---
+
 -- | See 'Streamly.Internal.Data.Parser.choice'.
 --
--- /Unimplemented/
+-- /Internal/
 --
 {-# INLINE choice #-}
-choice ::
-    -- Foldable t =>
-    t (Parser m a b) -> Parser m a b
-choice _ps = undefined
+choice :: (MonadCatch m, Foldable t) => t (Parser m a b) -> Parser m a b
+choice = foldl1 shortest
```

Comment:
`shortest` is marked as `Broken`, so we need to mark this also broken unless we review and fix that.